src/auth/hooks.ts

```diff
@@ -268,22 +258,18 @@ export const useAuthSetting = (storage: Storage, options?: UseQueryOptions<AuthS
   return query.data
 }
 
-export const getAuthSetting = async (storage: Storage) => {
-  const authSetting = parseAuthSetting(await storage.getItem(SettingsStorageKeys.Auth))
-  if (isAuthSetting(authSetting)) return authSetting
+export const getAuthSetting = async (storage: YoroiStorage) => {
+  const authSetting = await storage.join('appSettings/').getItem('auth', parseAuthSetting)
+  if (authSetting !== null) return authSetting
   return Promise.reject(new Error('useAuthSetting invalid data'))
```

Comment:
how does the app handle invalid auth setting?
is there a sane default value?

Reply:
unreachable code at 264, it fallbacks to `undefined`, maybe this helper doesn't make sense anymore and can be moved within the query,  `() => storage.join('appSettings/').getItem('auth', parseAuthSetting)`